lib/model/QubitUser.php

```diff
@@ -74,8 +74,26 @@ public function setPassword($password)
 
   public static function generatePasswordHash($password)
   {
-    $hashAlgo = constant(sfConfig::get('app_password_hash_algorithm', 'PASSWORD_ARGON2I'));
-    $hashAlgoOptions = json_decode(sfConfig::get('app_password_hash_algorithm_options', '{}'), true);
+    $hashAlgoConstant = sfConfig::get('app_password_hash_algorithm', 'PASSWORD_ARGON2I');
+
+    // Check to make sure hashing constant specified in settings is defined
+    if (defined($hashAlgoConstant))
+    {
+      // Hashing constant is defined so use it and options specified in settings
+      $hashAlgo = constant($hashAlgoConstant);
+
+      $hashAlgoOptions = json_decode(
+        sfConfig::get('app_password_hash_algorithm_options', '{}'),
+        true
+      );
+    }
+    else
+    {
+      // Hashing constant specified in settings isn't defined: use default
+      // alghorithm (PASSWORD_DEFAULT, usually PASSWORD_BCRYPT) and options
+      $hashAlgo = PASSWORD_DEFAULT;
+      $hashAlgoOptions = [];
+    }
 
```

Comment:
Maybe I am being extra paranoid, but I would like to see this function throw an exception if `$hashAlgo` is empty, before calling `password_hash()`, to ensure existing passwords aren't destroyed.

Reply:
Sounds good... I'll add that!

Reply:
Awesome!